client/src/html.js

```diff
@@ -17,6 +17,44 @@ export default class HTML extends React.Component {
           {this.props.headComponents}
         </head>
         <body {...this.props.bodyAttributes}>
+          <script
+            dangerouslySetInnerHTML={{
+              __html: `
+              (function() {
+                var preferredTheme;
+                
+                window.__onThemeChange = function() {};
+                window.__onThemeChangeEditor = function() {};
+
+                function setTheme(newTheme) {
+                  window.__theme = newTheme;
+                  preferredTheme = newTheme;
+                  document.body.className = newTheme;
+                  window.__onThemeChange(newTheme);
+                  window.__onThemeChangeEditor(newTheme);
+                }
+
+                try {
+                  preferredTheme = localStorage.getItem('theme');
+                } catch (err) { }
+
+                window.__setPreferredTheme = function(newTheme) {
+                  setTheme(newTheme);
+                  try {
+                    localStorage.setItem('theme', newTheme);
+                  } catch (err) {}
+                }
```

Comment:
localStorage is a big NO. It is going to be shared amongst sessions and hence across users.

Reply:
That makes sense.
In that case gatsby-plugin-dark-mode is also out of the picture.
One benefit of using localStorage would be that we might be able to use it persist theme across our services.